src/main/java/com/bootcamp/demo/restapi/exceptionhandler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,115 @@
+package com.bootcamp.demo.restapi.exceptionhandler;
+
+import com.bootcamp.demo.service.exception.ItemNotFoundException;
+import com.bootcamp.demo.service.exception.ServiceException;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.TypeMismatchException;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GlobalExceptionHandler.class);
+
+    /*
+        Global exception handlers
+     */
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ApiError> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
+        List<String> errors = ex.getBindingResult()
+                .getFieldErrors()
+                .stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .collect(Collectors.toList());
+        String errorMessage = String.join(", ", errors);
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
+    }
+
+    @ExceptionHandler(NoHandlerFoundException.class)
```

Comment:
If you look into NoHandlerFoundException, you will see that it is thrown if throwExceptionIfNoHandlerFound property is set to true, otherwise the default app behaviour is to return 404 status code. 
Does wrapping this into a BadRequest status add any value?